scripts/js/lib/api/releaseNotes.ts

```diff
@@ -257,3 +272,53 @@ async function writeReleaseNoteForVersion(
     await writeFile(versionPath, `${initialHeader}\n## ${versionsMarkdown}`);
   }
 }
+
+export function groupByMajorVersion(
+  versions: string[],
+): Record<string, string[]> {
+  const grouped = groupBy(versions, (v) => {
+    const match = v.match(/\[(\d+\.\d+)\]/);
+    if (!match || !match[1]) return "unknown";
+
+    const versionNumber = parseFloat(match[1]);
+    if (isNaN(versionNumber)) return "unknown";
+
+    return Math.floor(versionNumber).toString();
+  });
+
+  const sortedKeys = Object.keys(grouped)
+    .filter((key) => key !== "unknown")
+    .map(Number)
+    .sort((a, b) => b - a)
+    .map(String);
+
+  const sortedRecord: Record<string, string[]> = {};
+  for (const key of sortedKeys) {
+    const items = grouped[key];
+    if (Array.isArray(items)) {
```

Comment:
This if clause shouldn't be necessary and TypeScript should already have the type of `items` as `string[]`. Were you getting issues with TypeScript? Or was this perhaps AI suggesting the code?